internal/cloud_connector/utils.go

```diff
@@ -0,0 +1,28 @@
+package cloud_connector
+
+import (
+	"github.com/google/uuid"
+)
+
+func sanitizeCanonicalFacts(cf interface{}) interface{} {
+
+	canonicalFacts := cf.(map[string]interface{})
+	sanitizedCanonicalFacts := make(map[string]interface{})
+
+	for key, value := range canonicalFacts {
+		if key == "insights_id" {
```

Comment:
I think this reads better as a one liner?
```suggestion
		if (key == "insights_id") &&  (validUUID(value.(string))) {
```

Reply:
Good call.  I'm going to push this to get things moving and then I'll come back and make that change.  Thanks!